### PR TITLE
#24: [New Rule] Static methods SHOULD NOT be used

### PR DESCRIPTION
**Pull request by larsroettig**
#24

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento-coding-standard?pullRequest=41) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento-coding-standard?pullRequest=41) before we can accept your contribution.<br/>

File: Magento/Sniffs/Functions/StaticFunctionSniff.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright © Magento. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
```
Comment (lenaorobei) [inline Magento/Sniffs/Functions/StaticFunctionSniff.php]: New line is redundant here.

File: Magento/Sniffs/Functions/StaticFunctionSniff.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ * Detects static function definitions.
+ */
+class StaticFunctionSniff implements Sniff
+{
+    /**
+     * String representation of warning.
+     *
+     * @var string
+     */
+    protected $warningMessage = 'The use of `static` methods is discouraged';
```
Comment (lenaorobei) [inline Magento/Sniffs/Functions/StaticFunctionSniff.php]: Could you please add more explanatory message? Something like:
> Static method cannot be intercepted and its use is discouraged.

File: Magento/Sniffs/Functions/StaticFunctionSniff.php (hunk 1 above)
Comment (larsroettig) [reply Magento/Sniffs/Functions/StaticFunctionSniff.php]: done

**Pull request merged**